config/configmapprovider/retrieved.go

```diff
@@ -21,38 +21,42 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-// Retrieved holds the result of a call to the Retrieve method of a Provider object.
-// This interface cannot be directly implemented. Implementations must use the NewRetrieved helper.
-type Retrieved interface {
-	// Get returns the config Map. Should never be called after Close.
-	// Should never be called concurrently with itself or Close.
-	Get(ctx context.Context) (*config.Map, error)
-
-	// Close signals that the configuration for which it was used to retrieve values is
-	// no longer in use and should close and release any watchers that it may have created.
-	//
-	// Should block until all resources are closed, and guarantee that `onChange` is not
-	// going to be called after it returns except when `ctx` is cancelled.
-	//
-	// Should never be called concurrently with itself or Get.
-	Close(ctx context.Context) error
+// Get returns the config Map. Should never be called after Close.
+// Should never be called concurrently with itself or Close.
+//
+// Deprecated: Not needed, will be removed soon.
+func (r Retrieved) Get(ctx context.Context) (*config.Map, error) {
+	return r.Cfg, nil
+}
 
-	// privateRetrieved is an unexported func to disallow direct implementation.
-	privateRetrieved()
+// Close signals that the configuration for which it was used to retrieve values is
+// no longer in use and should close and release any watchers that it may have created.
+//
+// Should block until all resources are closed, and guarantee that `onChange` is not
+// going to be called after it returns except when `ctx` is cancelled.
+//
+// Should never be called concurrently with itself or Get.
+//
+// Deprecated: Not needed, will be removed soon.
+func (r Retrieved) Close(ctx context.Context) error {
+	if r.CloseFunc == nil {
+		return nil
+	}
+	return r.CloseFunc(ctx)
 }
 
 // GetFunc specifies the function invoked when the Retrieved.Get is being called.
+// Deprecated: Not needed, will be removed soon.
```

Comment:
Can you add a note for users to know why this is "Not needed" 😬

Reply:
:)) I don't know how to explain :)) more than you have access to the member...